contracts/CourtPresaleActivate.sol

```diff
@@ -49,10 +59,93 @@ contract CourtPresaleActivate is IsContract, ApproveAndCallFallBack {
+        ERC20 token = ERC20(_token);
+        require(token.safeTransferFrom(_from, address(this), _amount), ERROR_TOKEN_TRANSFER_FAILED);
 
+        _buyAndActivate(_from, _amount, _token);
+
+        // refund leftovers if any
+        _refund(token);
+    }
+
+    /**
+    * @dev This function needs a previous approval on the external token used for the contributed amount.
+    *      It will pull the approved tokens, convert them in Uniswap to the presale contribution token,
+    *      convert the obtained tokens in the presale instance, and activate the converted tokens in the
+    *      jurors registry instance of the Aragon Court.
+    * @param _token Address of the external contribution token used
+    * @param _amount Amount of contribution tokens to be converted and activated
+    * @param _minTokens Minimum amount of presale contribution tokens obtained in Uniswap
+    * @param _minEth Minimum amount of ETH obtained in Uniswap (Uniswap internally converts first to ETH and then to target token)
+    * @param _deadline Transaction deadline for Uniswap
+    */
+    function contributeExternalToken(
+        address _token,
+        uint256 _amount,
+        uint256 _minTokens,
+        uint256 _minEth,
+        uint256 _deadline
+    )
+        external
+    {
+        require(_amount > 0, ERROR_ZERO_AMOUNT);
+
+        // move tokens to this contract
+        require(ERC20(_token).safeTransferFrom(msg.sender, address(this), _amount), ERROR_TOKEN_TRANSFER_FAILED);
+
+        // get the Uniswap exchange for the contribution token
+        address payable uniswapExchangeAddress = uniswapFactory.getExchange(_token);
+        require(uniswapExchangeAddress != address(0), ERROR_UNISWAP_UNAVAILABLE);
+        IUniswapExchange uniswapExchange = IUniswapExchange(uniswapExchangeAddress);
+
+        // swap tokens
+        address contributionTokenAddress = address(presale.contributionToken());
+        ERC20 token = ERC20(_token);
+        require(token.safeApprove(address(uniswapExchange), _amount), ERROR_TOKEN_APPROVAL_FAILED);
+        uint256 contributionTokenAmount = uniswapExchange.tokenToTokenSwapInput(_amount, _minTokens, _minEth, _deadline, contributionTokenAddress);
+
+        // buy in presale
+        _buyAndActivate(msg.sender, contributionTokenAmount, contributionTokenAddress);
+
+        // refund leftovers if any
+        _refund(token);
+        _refund(contributionTokenAddress);
+    }
+
+    /**
+    * @dev It will send the received ETH to Uniswap to get presale contribution tokens,
+    *      convert the obtained tokens in the presale instance, and activate the converted tokens in the
+    *      jurors registry instance of the Aragon Court.
+    * @param _minTokens Minimum amount of presale contribution tokens obtained in Uniswap
+    * @param _deadline Transaction deadline for Uniswap
+    */
+    function contributeEth(uint256 _minTokens, uint256 _deadline) external payable {
```

Comment:
Perhaps it would be useful to expose a fallback function that calls this with `now` and `min = 1` like Uniswap does?